light_curves/code/sample_lc.py

```diff
@@ -0,0 +1,145 @@
+    get_hon_sample(coords, labels)  #2022MNRAS.511...54H
+    get_yang_sample(coords, labels)   #2018ApJ...862..109Y
+
+    # Variable AGN sample from Ranga/Andreas:
+    VAGN = pd.read_csv('data/WISE_MIR_variable_AGN_with_PS1_photometry_and_SDSS_redshift.csv')
+    vagn_coords = [SkyCoord(ra, dec, frame='icrs', unit='deg') for ra, dec in zip(VAGN['SDSS_RA'], VAGN['SDSS_Dec'])]
+    vagn_labels = ['WISE-Variable' for ra in VAGN['SDSS_RA']]
+    coords.extend(vagn_coords)
+    labels.extend(vagn_labels)    
+
+    #now get some "normal" QSOs for use in the classifier
+    #there are ~500K of these, so choose the number based on
+    #a balance between speed of running the light curves and whatever 
+    #the ML algorithms would like to have
+    num_normal_QSO = 2000
+    get_SDSS_sample(coords, labels, num_normal_QSO)
+
+    ## ADD TDEs to the sample, manually copied the TDE ZTF names from Hammerstein et al. 2023
+    tde_names = ['ZTF18aabtxvd','ZTF18aahqkbt','ZTF18abxftqm','ZTF18acaqdaa','ZTF18acpdvos','ZTF18actaqdw','ZTF19aabbnzo','ZTF18acnbpmd','ZTF19aakiwze','ZTF19aakswrb','ZTF17aaazdba','ZTF19aapreis','ZTF19aarioci','ZTF19abhejal','ZTF19abhhjcc','ZTF19abidbya','ZTF19abzrhgq','ZTF19accmaxo','ZTF20aabqihu','ZTF19acspeuw','ZTF20aamqmfk','ZTF18aakelin','ZTF20abjwvae','ZTF20abfcszi','ZTF20abefeab','ZTF20abowque','ZTF20abrnwfc','ZTF20acitpfz','ZTF20acqoiyt', 'ZTF20abnorit']
+    TDE_id2coord(tde_names,coords,labels)
+    
+
+    get_paper_sample('2015ApJ...810...14A','FermiBL',coords,labels)
+    get_paper_sample('2019A&A...627A..33D','Cicco19',coords,labels)
+    get_paper_sample('2022ApJ...933...37W','Galex variable 22',coords,labels)
+    get_paper_sample('2020ApJ...896...10B','Palomar variable 20',coords,labels)
+
+    #To remove duplicates from the list if combining multiple references clean_sample can be used 
+    # the call below with nonunique_sample just changes the structure to mimic the output of clean sample
+    coords_list, labels_list = nonunique_sample(coords, labels) 
+    print('final sample: ',len(coords))
+    return coords_list,labels_list
+ 
+
+def parallel_lc(coords_list,labels_list,parquet_savename = 'data/df_lc_.parquet.gzip'):
+    ''' Check all the archives for the light curve data of the 
+    list of coordinates given in input in parallel and return a 
+    muldidimensional lightcurve dataframe.'''
+    
+    max_fermi_error_radius = str(1.0)  
+    max_sax_error_radius = str(3.0)
+    heasarc_cat = ["FERMIGTRIG", "SAXGRBMGRB"]
+    error_radius = [max_fermi_error_radius , max_sax_error_radius]
+    bandlist = ["W1", "W2"]
+    wise_radius = 1.0 * u.arcsec
+    panstarrs_radius = 1.0 / 3600.0  # search radius = 1 arcsec
+    lk_radius = 1.0  # arcseconds
+    hcv_radius = 1.0 / 3600.0  # radius = 1 arcsec
+    n_workers = 8  # this should equal the total number of archives called
+    ztf_nworkers = None
+
+    parallel_starttime = time.time()
+
+    # start a multiprocessing pool and run all the archive queries
```

Comment:
```suggestion
    # start a multiprocessing pool and run all the archive queries
    # "spawn" new processes because it uses less memory and is thread safe
    # in particular, this is required for pd.read_parquet (used by ZTF_get_lightcurve)
    # https://stackoverflow.com/questions/64095876/multiprocessing-fork-vs-spawn
    mp.set_start_method("spawn", force=True)
```